File: supabase/functions/_shared/brevo.ts
// Shared Brevo helper for sending transactional emails via Brevo's REST API.
// Uses fetch (Deno-native) instead of @getbrevo/brevo (Node SDK) for compatibility.

const BREVO_API_URL = "https://api.brevo.com/v3/smtp/email";

const SENDER = { name: "Danielle from Cobbli", email: "noreply@cobbli.com" };
const REPLY_TO = { email: "support@cobbli.com" };

export interface SendBrevoEmailOptions {
  templateId: number;
  to: { email: string; name?: string }[];
  params?: Record<string, unknown>;
  tags?: string[];
}

export async function sendBrevoEmail(opts: SendBrevoEmailOptions) {
  const apiKey = Deno.env.get("BREVO_API_KEY");
  if (!apiKey) throw new Error("BREVO_API_KEY is not configured");

  const payload = {
    sender: SENDER,
    replyTo: REPLY_TO,
    to: opts.to,
    templateId: opts.templateId,
    params: opts.params ?? {},
    tags: opts.tags,
  };

  const res = await fetch(BREVO_API_URL, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "api-key": apiKey,
      accept: "application/json",
    },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    const text = await res.text();
    throw new Error(`Brevo send failed (${res.status}): ${text}`);
  }
  return res.json();
}

export const corsHeaders = {
  "Access-Control-Allow-Origin": "*",
  "Access-Control-Allow-Headers":
    "authorization, x-client-info, apikey, content-type",
};

File: backend/app/services/brevo.py
from __future__ import annotations

import httpx

from app.settings import get_settings

BREVO_API_URL = "https://api.brevo.com/v3/smtp/email"
SENDER = {"name": "Danielle from Cobbli", "email": "noreply@cobbli.com"}
REPLY_TO = {"email": "support@cobbli.com"}


class BrevoError(RuntimeError):
    pass


async def send_brevo_email(
    *,
    to: list[dict[str, str]],
    template_id: int | None = None,
    subject: str | None = None,
    html_content: str | None = None,
    text_content: str | None = None,
    params: dict | None = None,
    tags: list[str] | None = None,
) -> dict:
    s = get_settings()
    if not s.brevo_api_key:
        raise BrevoError("BREVO_API_KEY not configured")

    payload: dict = {"sender": SENDER, "replyTo": REPLY_TO, "to": to}
    if tags:
        payload["tags"] = tags
    if template_id is not None:
        payload["templateId"] = template_id
        payload["params"] = params or {}
    if subject is not None:
        payload["subject"] = subject
    if html_content is not None:
        payload["htmlContent"] = html_content
    if text_content is not None:
        payload["textContent"] = text_content

    async with httpx.AsyncClient(timeout=15.0) as client:
        res = await client.post(
            BREVO_API_URL,
            json=payload,
            headers={
                "Content-Type": "application/json",
                "api-key": s.brevo_api_key,
                "accept": "application/json",
            },
        )
    if res.status_code >= 400:
        raise BrevoError(f"Brevo send failed ({res.status_code}): {res.text}")
    return res.json()
